cowamm/cow_amm_cow_with_users_surplus_query_4031724.sql

```diff
@@ -19,38 +18,92 @@ with cow_amm_surplus as (
-    inner join cow_protocol_{{blockchain}}.batches as b on cow_per_batch.tx_hash = b.tx_hash
+    inner join cow_protocol_ethereum.batches as b on cow_per_batch.tx_hash = b.tx_hash
 ),
 
-aggregate_results_per_solver as (
+cow_surplus_per_batch_gnosis as (
+    select
+        cow_per_batch.block_time,
+        cow_per_batch.tx_hash,
+        solver_address,
+        naive_cow,  -- fraction of batch volume traded within a CoW
+        surplus as surplus_in_usd,  -- surplus of the executed CoW AMM order, expressed in USD
+        naive_cow * surplus as realized_cow_surplus_in_usd -- surplus of the CoW AMM that is assumed to be generated via a CoW.
+    from "query_4025739(blockchain='gnosis',start_time='{{start_time}}',end_time='{{end_time}}')" as cow_per_batch
+    inner join cow_amm_surplus on cow_per_batch.tx_hash = cow_amm_surplus.tx_hash
+    inner join cow_protocol_gnosis.batches as b on cow_per_batch.tx_hash = b.tx_hash
+),
+
+aggregate_results_per_solver_ethereum as (
+    select
+        name as solver_name,
+        sum(realized_cow_surplus_in_usd) as total_cow_surplus_in_usd
+    from cow_surplus_per_batch_ethereum
+    inner join cow_protocol_ethereum.solvers as s on cow_surplus_per_batch_ethereum.solver_address = s.address and s.active
+    group by name
+),
+
+aggregate_results_per_solver_gnosis as (
     select
         name as solver_name,
         sum(realized_cow_surplus_in_usd) as total_cow_surplus_in_usd
-    from cow_surplus_per_batch
-    inner join cow_protocol_{{blockchain}}.solvers as s on cow_surplus_per_batch.solver_address = s.address and s.active
+    from cow_surplus_per_batch_gnosis
+    inner join cow_protocol_gnosis.solvers as s on cow_surplus_per_batch_gnosis.solver_address = s.address and s.active
     group by name
 ),
 
+aggregate_results_per_solver_all_chains_temp as (
+    select *
+    from aggregate_results_per_solver_ethereum
+    union all
+    select *
+    from aggregate_results_per_solver_gnosis
+),
+
+aggregate_results_per_solver_all_chains as (
+    select
+        solver_name,
+        sum(total_cow_surplus_in_usd) as total_cow_surplus_in_usd
+    from
+        aggregate_results_per_solver_all_chains_temp
+    group by solver_name
+),
+
 total_surplus as (
-    select sum(total_cow_surplus_in_usd) as total_surplus_in_usd from aggregate_results_per_solver
+    select sum(total_cow_surplus_in_usd) as total_surplus_in_usd from aggregate_results_per_solver_all_chains
 ),
 
-final_results_per_solver as (
+final_results_per_solver_prelim as (
     select
         arps.solver_name,
         total_cow_surplus_in_usd,
-        {{budget}} * arps.total_cow_surplus_in_usd / ts.total_surplus_in_usd as total_cow_reward
-    from aggregate_results_per_solver as arps cross join total_surplus as ts
+        {{cow_budget}} * arps.total_cow_surplus_in_usd / ts.total_surplus_in_usd as total_cow_reward
+    from aggregate_results_per_solver_all_chains as arps cross join total_surplus as ts
+),
+
+named_results as (
```

Comment:
nit

```suggestion
reward_targets as (
```

is clearer imo (generally there are a lot of little meaning table names in this query like temp, final, results etc)